eps/renderer.go

```diff
@@ -78,16 +83,29 @@ func (r *EPS) RenderPath(path *Path, style Style, m Matrix) {
 	r.w.Write([]byte(" fill"))
 }
 
-func (r *EPS) RenderText(text *Text, m Matrix) {
+func (r *Renderer) RenderText(text *canvas.Text, m canvas.Matrix) {
 	// TODO: (EPS) write text natively
 	paths, colors := text.ToPaths()
 	for i, path := range paths {
-		style := DefaultStyle
+		style := canvas.DefaultStyle
 		style.FillColor = colors[i]
 		r.RenderPath(path, style, m)
 	}
 }
 
-func (r *EPS) RenderImage(img image.Image, m Matrix) {
+func (r *Renderer) RenderImage(img image.Image, m canvas.Matrix) {
 	// TODO: (EPS) write image
 }
+
+type dec float64
+
+func (f dec) String() string {
```

Comment:
This could be replaced by a property of the `Renderer`:
```go
type Renderer struct {
	w             io.Writer
	width, height float64
	color         color.RGBA
        Float64Text  func(float64) string
}
```
With an exported function on `canvas`:
```go
func Float64Text(precision int) func(float64) string {
	return func(f float64) string {
		s := fmt.Sprintf("%.*f", precision, f)
		s = string(minify.Decimal([]byte(s), precision))
		if float64(math.MaxInt32) < f || f < float64(math.MinInt32) {
			if i := strings.IndexByte(s, '.'); i == -1 {
				s += ".0"
			}
		}
		return s
	}
}
```

And initilized on the constructor:
```
	return &Renderer{
		w:      w,
		width:  width,
		height: height,
		color:  canvas.Black,
                Float64Text: canvas.Float64Text(canvas.Precision),
	}
```

Reply:
I think I'd prefer to either copy the `dec` code to the renderer sub-packages, or to make a proper struct out of both number formatters, perhaps placed in a separate library (such as `tdewolff/parse`). Perhaps a little copying is better than an extra dependency?

Reply:
You are right, we can copy it to subpackages first and maybe refactor it in the future (if a lot of subpackages need this function, it can be moved to the root package, which is already a dependency).